system-addon/test/functional/mochitest/browser_topsites_dismissItem.js

```diff
@@ -0,0 +1,34 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+"use strict";
+
+/**
+ * Test verifies that the next top site in queue replaces a dismissed top site.
+ */
+
+test_newtab({
+  before: setDefaultTopSites,
+  test: async function defaultTopSites_dismiss() {
+    await ContentTaskUtils.waitForCondition(() => content.document.querySelector(".top-site-icon"),
+      "Topsite tippytop icon not found");
+
+    let defaultTopSitesNumber = content.document.querySelector(".top-sites-list").querySelectorAll("[class=\"top-site-outer\"]").length;
+    Assert.equal(defaultTopSitesNumber, 6, "6 top sites are loaded by default");
+
+    let firstTopSite = content.document.querySelector(".top-sites-list li:first-child a").getAttribute("href");
+    Assert.equal(firstTopSite, "https://www.youtube.com/");
```

Comment:
Let's not test for exact string matches. We can get the `href` of the second topsite and check that it moved to the first position.